app/views/teams/_solved_challenges_table.html.haml

```diff
@@ -4,9 +4,14 @@
       %th= t('teams.summary.solved_challenges_table.challenge_header')
       %th= t('teams.summary.solved_challenges_table.points_header')
       %th= t('teams.summary.solved_challenges_table.when_header')
-  %tbody
-    - @team.solved_challenges.each_with_index do |s, i|
+    = render partial: 'teams/challenge_list_tbody', locals: { teams: @team.solved_challenges.slice(0..4), starting_rank: 1 }
+    / This part of the list is hidden until the user clicks the "Show/Hide" button below
+    = render partial: 'teams/challenge_list_tbody', locals: { |
+             teams: @team.solved_challenges.slice(5..),       |
+             starting_rank: 6,                                |
```

Comment:
`starting_rank` as a variable name doesn't really make sense as the variable name here, more appropriate would be `starting_index` or something else, `starting_rank` really only made sense for the teams list.